packages/fondue/src/components/Tree/TreeItem/TreeItemMultiselect.tsx

```diff
@@ -139,7 +139,11 @@ export const TreeItemMultiselect = memo(
                         label=""
                         onChange={checkBoxOnSelect}
                         size={CheckboxSize.Default}
-                        state={getMultiselectCheckBoxState(isSelected, isPartialSelected)}
+                        state={
+                            isDisabled
+                                ? CheckboxState.Unchecked
+                                : getMultiselectCheckBoxState(isSelected, isPartialSelected)
+                        }
```

Comment:
Thanks @anxobotana 

I think I understand better now.

My only concern with this change is that it _always_ unchecks the Checkbox when node is disabled.

What if we have an initial state of both _checked_ and _disabled_? That would be valid but this logic would reset the state to unchecked.

Reply:
I understand your concern. I don't think your suggestion will work, since the selected state is calculated in the `useMultiselectTreeItem` and then this selected state propagates up and down in the tree. 

What about adding a new prop where the consumer can choose between both behaviors: `uncheckOnDisabled` default to `false`. 
and then
```suggestion
                          isDisabled && uncheckOnDisabled
                                ? CheckboxState.Unchecked
                                : getMultiselectCheckBoxState(isSelected, isPartialSelected)
```

Reply:
Sorry @anxobotana for the confusion, mine was a question not a suggestion 😄 

I'll take a look into code to get more familiar with the logic and will share with you any findings

Reply:
No problem at all, @floroz 

Thanks for looking into this